### PR TITLE
Adding contentType to document

### PR DESCRIPTION
**Pull request by pradeepr-roboticist**
## Motivation and Context (Why the change? What's the scenario?)

This addresses #471. 

## High level description (Approach, Design)

I just added a `contentType` to the document. Seems to fix some problems I faced with #471.

### CODE REVIEW

File: extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs [hunk 1]
```diff
@@ -105,7 +107,8 @@ public Task DeleteDocumentDirectoryAsync(string index, string documentId, Cancel
                 {
                     { "index", index },
                     { "documentId", documentId },
-                    { "fileName", fileName }
+                    { "fileName", fileName },
+                    { "contentType", "application/octet-stream"}
```
Comment (dluc) [inline extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs:111]: We should reuse mime detection, the approach used in other content storage classes, deriving the type from the file name 

e.g. see https://github.com/microsoft/kernel-memory/blob/dcd2a0e9c5dfeb1f896f746f21272673f71a4109/extensions/AzureBlobs/AzureBlobsStorage.cs#L196

File: extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs [hunk 2]
```diff
@@ -78,7 +78,8 @@ public Task DeleteDocumentDirectoryAsync(string index, string documentId, Cancel
                 { "_id", id },
                 { "documentId", documentId },
                 { "fileName", fileName },
-                { "content", new BsonString(await reader.ReadToEndAsync(cancellationToken).ConfigureAwait(false)) }
+                { "content", new BsonString(await reader.ReadToEndAsync(cancellationToken).ConfigureAwait(false)) },
+                { "contentType", "text/plain"}
```
Comment (dluc) [inline extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs:82]: nit: use `MimeTypes.PlainText`

File: extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs [hunk 3]
```diff
@@ -94,6 +95,7 @@ public Task DeleteDocumentDirectoryAsync(string index, string documentId, Cancel
             doc["documentId"] = documentId;
             doc["fileName"] = fileName;
             doc["content"] = content;
+            doc["contentType"] = "text/plain";
```
Comment (dluc) [inline extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs:98]: nit: use `MimeTypes.PlainText`

**Review by dluc — CHANGES_REQUESTED** [review]
LGTM, I would use mime detection though, it's free :-)

File: extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs (hunk 1 above)
Comment (pradeepr-roboticist) [reply extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs]: Thanks! Done!

File: extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs (hunk 2 above)
Comment (pradeepr-roboticist) [reply extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs]: Done!

File: extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs (hunk 3 above)
Comment (pradeepr-roboticist) [reply extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs]: Done!

File: extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs [hunk 4]
```diff
@@ -105,7 +107,8 @@ public Task DeleteDocumentDirectoryAsync(string index, string documentId, Cancel
                 {
                     { "index", index },
                     { "documentId", documentId },
-                    { "fileName", fileName }
+                    { "fileName", fileName },
+                    { "contentType", (new Pipeline.MimeTypesDetection()).GetFileType(fileName)}
```
Comment (dluc) [inline extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs:111]: this hardcodes the MimeTypesDetection class, which blocks from injecting a different implementation. Could you use DI, injecting the dependency in the ctor? You can reuse the code from `AzureBlobsStorage`

image

Comment (pradeepr-roboticist) [reply extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasStorage.cs]: Got it. Thanks for the reference.

**Review by dluc — APPROVED** [review]

**Comment by dluc** [comment]
hi @pradeepr-roboticist the PR is ready for merge, only pending CLA, see the comments above. Thanks!

**Comment by pradeepr-roboticist** [comment]
> please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.

@microsoft-github-policy-service agree

**Comment by pradeepr-roboticist** [comment]
> hi @pradeepr-roboticist the PR is ready for merge, only pending CLA, see the comment above. Thanks!

Just agreed to CLA. Thanks for the quick review!

**Pull request merged**